src/metabase/lib/schema/literal.cljc

```diff
@@ -14,23 +15,88 @@
 (mr/def ::string
   [:string])
 
+;;; TODO -- these temporal literals could be a little stricter, right now they are pretty permissive, you shouldn't be
+;;; allowed to have month `13` or `02-29` for example
+
+(def ^:private date-part-regex #"\d{4}-\d{2}-\d{2}")
```

Comment:
Added better schemas for all of the different temporal literal strings we support. I also added support for `java.time.*` literals, only in Clojure. Malli had support for this stuff in the `malli.experimental.time` namespace; I added that to our default schema registry. Those schemas are the `:time/*` ones you see below.